util/keydecoder/keydecoder.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keydecoder
+
+import (
+	"fmt"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+// HandleType is the type of a Handle, could be `int`(ie. kv.IntHandle) or `common`(ie. *kv.CommonHandle)
+type HandleType string
+
+// HandleType instances
+const (
+	IntHandle    HandleType = "int"
+	CommonHandle HandleType = "common"
+
+	UnknownHandle HandleType = "unknown"
+)
+
+// DecodedKey is a struct contains detailed information about a key, its json form should be used to fill KEY_INFO field in `DEADLOCKS` and `DATA_LOCK_WAITS`
+type DecodedKey struct {
+	DbID              int64      `json:"db_id,omitempty"`
+	DbName            string     `json:"db_name,omitempty"`
+	TableID           int64      `json:"table_id"`
+	TableName         string     `json:"table_name,omitempty"`
+	HandleType        HandleType `json:"handle_type,omitempty"`
+	IsPartitionHandle bool       `json:"partition_handle,omitempty"`
```

Comment:
This won't be displayed in json if `IsPartitionHandle` is false (cause it is the default value).
However I don't know this behaviour is preferred…

Reply:
I think it doesn't matter.